src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,457 @@
+     *
+     * @param variableDef DetailAst of type {@link TokenTypes#VARIABLE_DEF}
+     * @return scope of variableDef
+     */
+    private static DetailAST findScopeOfVariable(DetailAST variableDef) {
+        final DetailAST result;
+        final DetailAST parentAst = variableDef.getParent();
+        if (parentAst.getType() == TokenTypes.SLIST) {
+            result = parentAst;
+        }
+        else {
+            final DetailAST grandParent = parentAst.getParent();
+            final DetailAST scopeAst = grandParent.findFirstToken(TokenTypes.SLIST);
+            if (scopeAst == null) {
+                result = grandParent;
+            }
+            else {
+                result = scopeAst;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is
+     * used as left-hand side value. An identifier is being used as a left-hand side
+     * value if it is used as the left operand of an assignment or as an
+     * operand of a stand-alone increment or decrement.
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return True if identAst is used as a left-hand side value otherwise false.
+     */
+    private static boolean isLeftHandSideValue(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        return isStandAloneIncrementOrDecrement(identAst)
+                || parent.getType() == TokenTypes.ASSIGN
+                && identAst != parent.getLastChild();
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is used as
+     * an operand of a stand-alone increment or decrement.
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return True if identAst is used as an operand of stand-alone
+     *         increment or decrement otherwise false
+     */
+    private static boolean isStandAloneIncrementOrDecrement(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        final DetailAST grandParent = parent.getParent();
+        return TokenUtil.isOfType(parent, INCREMENT_AND_DECREMENT_TOKENS)
+                && TokenUtil.isOfType(grandParent, TokenTypes.EXPR)
+                && (!TokenUtil.isOfType(grandParent.getParent(),
+                TokenTypes.ELIST, TokenTypes.INDEX_OP, TokenTypes.ASSIGN)
+                || grandParent.getParent().getParent().getType() == TokenTypes.FOR_ITERATOR);
+    }
+
+    /**
+     * Maintains information about the local variable.
+     */
+    private static final class VariableDesc {
+
+        /**
+         * The name of the variable.
+         */
+        private final String name;
+
+        /**
+         * DetailAst of type {@link TokenTypes#TYPE}.
+         */
+        private final DetailAST typeAst;
+
+        /**
+         * The scope of variable is determined by the DetailAst of type
+         * {@link TokenTypes#SLIST} which is enclosing the variable.
```

Comment:
Seems to me that this could also be a `LITERAL_FOR`, by check logic above.

Reply:
yes, `LITERAL_FOR` can be the parent, updated JavaDoc, `RESOURCES` can't be, for reason mentioned below.

Reply:
Resolved.